codeclimate-wrapper

```diff
@@ -41,10 +41,20 @@ else
     && invalid_docker_host "$DOCKER_HOST"
 fi
 
-exec docker run \
+if [ -t 1 ]; then
+  exec docker run \
   --interactive --tty --rm \
   --env CODE_PATH="$PWD" \
   --volume "$PWD":/code \
   --volume /tmp/cc:/tmp/cc \
   --volume /var/run/docker.sock:/var/run/docker.sock \
   codeclimate/codeclimate "$@"
+else
+  exec docker run \
+    --rm \
+    --env CODE_PATH="$PWD" \
+    --volume "$PWD":/code \
+    --volume /tmp/cc:/tmp/cc \
+    --volume /var/run/docker.sock:/var/run/docker.sock \
+    codeclimate/codeclimate "$@"
```

Comment:
I'm wondering if it's worth DRYing this so the two don't drift out of since. One way to do that would be to exctract a function:

```sh
docker_run() {
  exec docker run --all --the --shared --options "$@"
}

if ...; then
  docker_run --interactive --tty codeclimate/codeclimate "$@"
else
  docker_run codeclimate/codeclimate "$@"
fi
```

WDYT?